estmator_project/estmator_project/views.py

```diff
@@ -17,23 +22,47 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class QuoteView(TemplateView):
-    template_name = 'quote.html'
+class AboutView(TemplateView):
+    template_name = 'about.html'
 
     def get_context_data(self, **kwargs):
-        try:
-            context = super(QuoteView, self).get_context_data(**kwargs)
-            context['categories'] = Category.objects.all()
+        context = super(AboutView, self).get_context_data(**kwargs)
+        return context
 
-            options_form = QuoteOptionsForm()
-            context['options_form'] = options_form.as_ul
 
-            context['quote_client'] = self.request.GET.get('client')
-            context['quote_name'] = self.request.GET.get('name')
-        except (KeyError, ValueError):
-            return redirect('menu')
+@login_required
+def quote_view(request):
+    if request.method == 'POST':
+        categories = {c: {p: 0 for p in c.product_set.all()} for c in Category.objects.all()}
+        if 'quote' in request.POST:
+            quote = Quote.objects.get(id=request.POST['quote'])
+            quote_name = quote.name
+            for prop in quote.productproperties_set.all():
+                categories[prop.product.category][prop.product] = prop.count
+        else:
+            quote = None
+            quote_name = request.POST['name']
 
-        return context
+        form_html = ''
+        options_form = QuoteOptionsForm(instance=quote)
+
+        for fieldname, field in options_form.fields.items():
+            form_html += '<li><a>'
+            form_html += str(options_form[fieldname]) + ' '
+            form_html += options_form[fieldname].label
+            form_html += '</a></li>'
```

Comment:
This method of building a form by hand is really antithetical to what Django is.  You should be able to epxress this form in such a way that you do not need to manually construct any html, but can simply render the form or its fields in the page template where it belongs:

```html
{{ form.as_p() }}
``` 
or
```html
{% for field in form %}
<!-- render the field individually here -->
{% endfor %}
```